src/Tests/Microsoft.Diagnostics.Monitoring.StartupHook.UnitTests/Exceptions/Eventing/ExceptionsEventListener.cs

```diff
@@ -115,6 +118,16 @@ private static ulong ToUInt64(object? value)
             return ToType<ulong>(value);
         }
 
+        private static T ToEnum<T>(object? value)
```

Comment:
What's the actual type of the value when this is called? We might be able to write somethingmore succinct here.

Reply:
Switched to using something specific for ActivityIdFormat - I can go back to a more generic method if we'd rather do that though.

Reply:
I'd still like to know what the runtime type of the `value` parameter is; there might be more efficient code that can be used knowing what the expected type is. Please use a debugger to get the actual type.

Reply:
Sorry, I thought you just wanted to reduce all the verbosity with T - the type is listed as `object {int}` according to the debugger

Reply:
With that known, it can be updated to something like:

```cs
if (value is int intValue)
{
  return (ActivityIdFormat)intValue;
}
throw new InvalidCastException();
```

No parsing or string conversion needed, just unboxing and casting.